### PR TITLE
Add basic math functions

### LINKED ISSUE #1: Include addition and summation functions
We should include `add/2`, `add!/2`, `sub/2`, `sub!/2`, `sum/1` and `sum_default/2` to Quantity. In addition we also need the `zero_value/1` helper.

### PR DESCRIPTION
**Pull request by skovmand**
Fixes #1

### CODE REVIEW

**Comment by skovmand** [comment]
@lasseebert I think this is ready for review when you have time.

File: lib/quantity/math.ex [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    result =
+      remaining
+      |> Enum.reduce_while(first, fn quantity, acc ->
+        case add(quantity, acc) do
+          {:ok, result} -> {:cont, result}
+          :error -> {:halt, :error}
+        end
+      end)
+
+    case result do
```
Comment (lasseebert) [inline lib/quantity/math.ex]: Tip: You can pipe into case

File: lib/quantity/math.ex [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  iex> sum([~Q[0.11 DKK], ~Q[0.22 DKK], ~Q[0.33 DKK]], -2, "DKK")
+  {:ok, ~Q[0.66 DKK]}
+
+  iex> sum([], 0, "DKK")
+  {:ok, ~Q[0 DKK]}
+
+  iex> sum([], -2, "DKK")
+  {:ok, ~Q[0.00 DKK]}
+
+  iex> sum([~Q[1 EUR], ~Q[2 DKK]], -2, "EUR")
```
Comment (lasseebert) [inline lib/quantity/math.ex]: Nice tests 👌

File: lib/quantity/math.ex [hunk 3]
```diff
@@ -0,0 +1,156 @@
+defmodule Quantity.Math do
+  @moduledoc """
+  Functions for doing math with Quantities
+  """
+
+  @doc """
+  Add two Quantities, keeping the unit
+
+  iex> add(~Q[1.34 MWh], ~Q[3.49 MWh])
+  {:ok, ~Q[4.83 MWh]}
```
[the remark this answers is not in the document]
Comment (skovmand) [reply lib/quantity/math.ex]: Good idea. I have added this one: 

```
iex> add(~Q[1.234567 days], ~Q[3.5 days])
{:ok, ~Q[4.734567 days]}
```

in https://github.com/wise-home/quantity/pull/2/commits/4e6374196946acc5a29039e0b39beab5d37fbc97

File: lib/quantity/math.ex (hunk 1 above)
Comment (skovmand) [reply lib/quantity/math.ex]: Oh of course :-) Thanks. Fixed in https://github.com/wise-home/quantity/pull/2/commits/3eea9c8a6c87bde09497be61491713b932bb8aa7

**Pull request merged**